src/internal_events/file.rs

```diff
@@ -478,6 +478,25 @@ mod source {
         }
     }
 
+    #[derive(Debug)]
+    pub struct GaveUpOnDeletedFile<'a> {
+        pub file: &'a Path,
+    }
+
+    impl<'a> InternalEvent for GaveUpOnDeletedFile<'a> {
+        fn emit(self) {
+            info!(
+                message = "Gave up on deleted file.",
+                file = %self.file.display(),
+            );
+            counter!(
+                "files_deleted_given_up_total", 1,
+                "file" => self.file.to_string_lossy().into_owned(),
+            );
+        }
+    }
```

Comment:
Can you please conditionally include the `file` tag on the `files_deleted_given_up_total` counter if and only if `include_file_metric_tag` (below) is true?

Reply:
On a another, higher-level note, I think it would be better to include this information as a tag on the `files_unwatched_total` counter rather than adding a new counter. The tag could be a boolean tag named something like `reached_eof`.